interop/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala

```diff
@@ -31,25 +54,175 @@ sealed abstract class CatsInstances2 {
+//    effect.IO.cancelable { cb =>
+//      runCancelable(fa)(r => effect.IO(cb(r))).unsafeRunSync()
+//      effect.IO(())
+//    }
+
+//    effect.IO.cancelable { cb =>
+//      runAsync(fa)(r => effect.IO(cb(r))).unsafeRunSync()
+//      unsafeRun(fa.run.flatMap(r => IO.sync(cb(exitResultToEither(r)))).fork)
+//    }
+
+//    effect.IO.cancelable { k =>
+//      val f = unsafeRun(fa.run.flatMap(r => IO.sync(k(exitResultToEither(r)))).fork)
+//      effect.IO(unsafeRun(f.interrupt))
+//    }
+
+}
+
+private class CatsConcurrent extends CatsEffect with Concurrent[Task] {
+  private[this] def toFiber[A](f: Fiber[Throwable, A]): effect.Fiber[Task, A] = new effect.Fiber[Task, A] {
+    override val cancel: Task[Unit] =
+      Task(System.out println "interrupt running") *>
+        f.interrupt.peek(_ => Task(System.out println "interrupt ran"))
+
+    override val join: Task[A] = f.join
+  }
+
+  override def liftIO[A](ioa: cats.effect.IO[A]): Task[A] =
+    Concurrent.liftIO(ioa)(this)
+
+  private[this] def toFiberFlatMapped[E1, E >: E1, A, B](
+    f: Fiber[E1, A],
+    flatMap: A => IO[E, B]
+  ): effect.Fiber[IO[E, ?], B] = new effect.Fiber[IO[E, ?], B] {
+    override val cancel: IO[E, Unit] =
+      IO.sync(System.out println "interruptFlatMapped running") *>
+        f.interrupt.peek(_ => IO.sync(System.out println "interruptFlatMapped ran"))
+
+    override val join: IO[E, B] = f.join.flatMap(flatMap)
+  }
+
+  override def cancelable[A](k: (Either[Throwable, A] => Unit) => effect.CancelToken[Task]): Task[A] =
+    IO.async0 { kk: Callback[Throwable, A] =>
+      System.out println "cancelable running"
+      val token = try {
+        k(e => kk(eitherToExitResult(e)))
+      } catch {
+        case e: Throwable =>
+          System.out println s"cancelable error $e"
+          throw e
+      }
+      val token0: Async[Nothing, A] = Async.maybeLater {
+        IO.sync(System.out println "cancel token running") *>
+          token.catchAll(IO.terminate)
+      }
+      System.out println "cancelable ran"
+      token0
+    }.peek(_ => Task(System.out println "cancelable finished"))
+
+  override def race[A, B](fa: Task[A], fb: Task[B]): Task[Either[A, B]] =
+    racePair(fa, fb).flatMap {
+      case Left((a, fiberB)) =>
+        fiberB.cancel.const(Left(a)).peek(_ => Task(System.out println "race: won A"))
+      case Right((fiberA, b)) =>
+        fiberA.cancel.const(Right(b)).peek(_ => Task(System.out println "race: won B"))
+    }.catchAll(e => Task(System.out println s"race: failed, got exception ${e.getMessage} ${stackTrace(e)}") *> IO.fail(e))
+
+  override def start[A](fa: Task[A]): Task[effect.Fiber[Task, A]] =
+    fa.fork.map(toFiber).peek(_ => Task(System.out println "start ran"))
+
+  override def racePair[A, B](
+    fa: Task[A],
+    fb: Task[B]
+  ): Task[Either[(A, effect.Fiber[Task, B]), (effect.Fiber[Task, A], B)]] =
+    (fa.sandboxed.attempt raceWith fb.sandboxed.attempt)(
```

Comment:
The implementation of this can probably be greatly simplified. 

In particular, `raceWith` now provides the full and complete `ExitStatus` of the first fiber to finish. Therefore, it's not necessary to sandbox / attempt the two sides.